src/rgw/rgw_rest_conn.h

```diff
@@ -67,8 +67,14 @@ inline param_vec_t make_param_list(const std::map<std::string, std::string> *pp)
 
 class RGWRESTConn
 {
+  struct conn_status {
+    bool connectable;
+    ceph::real_time last_status_update;
+  };
+
   CephContext *cct;
   std::vector<std::string> endpoints;
+  std::unordered_map<std::string, conn_status> endpoints_status;
```

Comment:
This  `RGWRESTConn` change is not necessary to resolve the radosgw-admin case, but make the logic (not only in radosgw-admin case) more efficient by avoiding some unnecessary retries.

This extra `endpoints_status` is only populated in the constructor of `RGWRESTConn` and the map won't be manipulated afterward. It's true that the status of each endpoint would be updated but I understood that it would be handled asyncly via coroutine and would not be accessed simultaneously in multi-thread env for each endpoint? I haven't dug thoroughly into the bottom of the code yet so my understanding might be wrong here..

Reply:
> I haven't dug thoroughly into the bottom of the code yet so my understanding might be wrong here..

i think you're right here. i was worried that metadata sync and data sync may share the same `RGWRESTConn` instance for the master zone, but i see that `RGWSI_Zone` creates a separate `rest_master_conn` for metadata sync

that said, i think it would be better for each sync thread to manage its own `RGWRESTConn` instance explicitly. these could be member variables of `RGWDataSyncStatusManager` and `RGWMetaSyncStatusManager`/`RGWRemoteMetaLog` instead of pointers; they'd only need to read the 'global' list of zone endpoints from `RGWSI_Zone` on construction

that encapsulation would help to prevent us from adding other thread-unsafe accesses to the members of `RGWSI_Zone`, and we could avoid the use of atomics in `RGWRESTConn` by documenting that it's not thread-safe

Reply:
Yup this makes sense to me. I'll make the changes in a separate PR (as you suggested in an earlier comment) since it's not a necessary change to solve this radosgw-admin sync status issue.

Reply:
After reading the code more carefully, I do notice that `rest_master_conn` is actually accessed by multiple threads: metadata sync thread, data sync thread, sync log trim thread, and rgw requests handling threads. And `RGWRESTConn` instance for each zone is accessed by multiple threads too. 
I'll put this part in a separate PR and discuss what's the best way to handle this there.

Reply:
This change will be submitted as part of the PR for the issue https://tracker.ceph.com/issues/62710